flytekit/annotated/workflow.py

```diff
@@ -152,9 +152,12 @@ def _local_execute(self, ctx: FlyteContext, **kwargs) -> Union[Tuple[Promise], P
             else:
                 raise Exception("something returned from wf but shouldn't have outputs")
 
-        if len(output_names) != len(function_outputs):
-            # Length check, clean up exception
-            raise Exception(f"Length difference {len(output_names)} {len(function_outputs)}")
+        if len(output_names) > 1 and isinstance(function_outputs, tuple):
+            if len(output_names) != len(function_outputs):
+                # Length check, clean up exception
+                raise Exception(f"Length difference {len(output_names)} {len(function_outputs)}")
+        elif len(output_names) == 1 and isinstance(function_outputs, Promise):
+            return function_outputs
```

Comment:
is the stuff below not needed?

Reply:
what do you mean? its like elif right? the indentation here looks weird, but seems fine in the code

Reply:
But it returns here right, so the stuff below won't run.  Should it not run?  I added some Promise renaming stuff but maybe it's not needed?

Reply:
done